src/wazuh_db/helpers/wdb_agents_helpers.c

```diff
@@ -181,3 +199,122 @@ int wdb_agents_vuln_cve_update_status(int id,
 
     return result;
 }
+
+cJSON* wdb_agents_vuln_cves_remove_by_status(int id,
+                                            const char *status,
+                                            int *sock) {
+    cJSON *data_in = NULL;
+    char *data_in_str = NULL;
+    char *wdbquery = NULL;
+    char *wdboutput = NULL;
+    char *payload = NULL;
+    cJSON *data_out = NULL;
+    wdbc_result wdb_res = WDBC_DUE;
+    int aux_sock = -1;
+
+    data_in = cJSON_CreateObject();
+
+    if (!data_in) {
+        mdebug1("Error creating data JSON for Wazuh DB.");
+        return data_out;
+    }
+
+    cJSON_AddStringToObject(data_in, "status", status);
+
+    data_in_str = cJSON_PrintUnformatted(data_in);
+    os_malloc(WDBQUERY_SIZE, wdbquery);
+    snprintf(wdbquery, WDBQUERY_SIZE, agents_db_commands[WDB_AGENTS_VULN_CVES_REMOVE], id, data_in_str);
+
+    os_malloc(WDBOUTPUT_SIZE, wdboutput);
+    while (wdb_res == WDBC_DUE) {
+        // Query WazuhDB
+        if (wdbc_query_ex(sock?sock:&aux_sock, wdbquery, wdboutput, WDBOUTPUT_SIZE) == 0) {
+            wdb_res = wdbc_parse_result(wdboutput, &payload);
+
+            if (WDBC_OK == wdb_res || WDBC_DUE == wdb_res) {
+                const char *error = NULL;
+                cJSON *cves = cJSON_ParseWithOpts(payload, &error, TRUE);
+
+                if (!cves) {
+                    mdebug1("Invalid vuln_cves JSON results syntax after removing vulnerabilities.");
+                    mdebug2("JSON error near: %s", error);
+                    wdb_res = WDBC_ERROR;
+                }
+                else if (!data_out) {
+                    data_out = cves;
```

Comment:
Maybe we can add a comment here explaining that the first time a chunk is received we only consider the query response as the response of the method.
We can also bring this format to the if statements:
```
if (!cves) {
}
else{
  //First time
  if(!dataout){
  }
  //Extra times
  else{
  }
}
```
Just to clarify that both of the last cases are for valid responses.

Reply:
I agree with you... Fixed.